src/transformers/integrations/peft.py

```diff
@@ -333,7 +356,34 @@ def active_adapter(self) -> str:
 
         for _, module in self.named_modules():
             if isinstance(module, BaseTunerLayer):
-                return module.active_adapter
+                active_adapters = module.active_adapter
+                break
+
+        # For previous PEFT versions
+        if isinstance(active_adapters, str):
+            active_adapters = [active_adapters]
+
+        return active_adapters
+
+    def active_adapter(self) -> str:
```

Comment:
I do not think it's a good idea to have both `active_adapter` and `active_adapters`:
- It's easier for the user to just have to remember one method and deal with the output type (we can also deprecate returning a single string in favor of returning a list going forward)
- Reduces the API surface - only a single source of truth